File: pkg/dev_compiler/test/codegen_expected/corelib/iterable_single_test.js
dart_library.library('corelib/iterable_single_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__iterable_single_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const _interceptors = dart_sdk._interceptors;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const iterable_single_test = Object.create(null);
  let JSArrayOfint = () => (JSArrayOfint = dart.constFn(_interceptors.JSArray$(core.int)))();
  let SetOfint = () => (SetOfint = dart.constFn(core.Set$(core.int)))();
  let VoidToint = () => (VoidToint = dart.constFn(dart.definiteFunctionType(core.int, [])))();
  let dynamicTobool = () => (dynamicTobool = dart.constFn(dart.definiteFunctionType(core.bool, [dart.dynamic])))();
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let const$;
  let const$0;
  let const$1;
  iterable_single_test.main = function() {
    let list1a = JSArrayOfint().of([1]);
    let list1b = JSArrayOfint().of([1, 2, 3]);
    let list1c = JSArrayOfint().of([]);
    let list2a = const$ || (const$ = dart.constList([5], core.int));
    let list2b = const$0 || (const$0 = dart.constList([4, 5], core.int));
    let list2c = const$1 || (const$1 = dart.constList([], core.int));
    let set1 = SetOfint().new();
    set1.add(22);
    let set2 = core.Set.new();
    set2.add(11);
    set2.add(12);
    set2.add(13);
    let set3 = core.Set.new();
    expect$.Expect.equals(1, list1a[dartx.single]);
    expect$.Expect.throws(dart.fn(() => list1b[dartx.single], VoidToint()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
    expect$.Expect.throws(dart.fn(() => list1c[dartx.single], VoidToint()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
    expect$.Expect.equals(5, list2a[dartx.single]);
    expect$.Expect.throws(dart.fn(() => list2b[dartx.single], VoidToint()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
    expect$.Expect.throws(dart.fn(() => list2c[dartx.single], VoidToint()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
    expect$.Expect.equals(22, set1.single);
    expect$.Expect.throws(dart.fn(() => set2.single, VoidTovoid()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
    expect$.Expect.throws(dart.fn(() => set3.single, VoidTovoid()), dart.fn(e => core.StateError.is(e), dynamicTobool()));
  };
  dart.fn(iterable_single_test.main, VoidTodynamic());
  // Exports:
  exports.iterable_single_test = iterable_single_test;
});

File: tests/corelib_strong/iterable_single_test.dart
// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

main() {
  List<int> list1a = <int>[1];
  List<int> list1b = <int>[1, 2, 3];
  List<int> list1c = <int>[];
  List<int> list2a = const <int>[5];
  List<int> list2b = const <int>[4, 5];
  List<int> list2c = const <int>[];
  Set<int> set1 = new Set<int>();
  set1..add(22);
  Set set2 = new Set();
  set2..add(11)
      ..add(12)
      ..add(13);
  Set set3 = new Set();

  Expect.equals(1, list1a.single);
  Expect.throws(() => list1b.single, (e) => e is StateError);
  Expect.throws(() => list1c.single, (e) => e is StateError);

  Expect.equals(5, list2a.single);
  Expect.throws(() => list2b.single, (e) => e is StateError);
  Expect.throws(() => list2c.single, (e) => e is StateError);

  Expect.equals(22, set1.single);
  Expect.throws(() => set2.single, (e) => e is StateError);
  Expect.throws(() => set3.single, (e) => e is StateError);
}
